scipy/fft/_backend.py

```diff
@@ -0,0 +1,169 @@
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+
+_named_backends = {
+    'scipy': _ScipyBackend,
+}
+
+def _backend_from_arg(backend):
+    """Maps strings to known backends and validates the backend"""
+
+    if isinstance(backend, str):
+        try:
+            backend = _named_backends[backend]
+        except KeyError:
+            raise ValueError('Unknown backend {}'.format(backend))
+
+    if backend.__ua_domain__ != 'numpy.scipy.fft':
+        raise ValueError('Backend does not implement "numpy.scipy.fft"')
+
+    return backend
+
+
+def set_global_backend(backend):
+    """Sets the global fft backend
+
+    The global backend has higher priority than registered backends, but lower
+    priority than context-specific backends set with `set_backend`.
+
+    Parameters
+    ----------
+    backend: {object, 'scipy'}
+
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'}, or an object that implements the uarray protocol.
+
+    Raises
+    ------
+    ValueError: If the backend does not implement ``numpy.scipy.fft``
+
+    Notes
+    -----
+    This will overwrite the previously set global backend, which by default is
+    the SciPy implementation.
+
+    """
+
+    backend = _backend_from_arg(backend)
+    ua.set_global_backend(backend)
+
+
+def register_backend(backend):
+    """
+    Register a backend for permanent use.
+
+    Registered backends have the lowest priority and will be tried after the
+    global backend.
+
+    Parameters
+    ----------
+    backend: {object, 'scipy'}
+
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'}, or an object that implements the uarray protocol.
+
+    Raises
+    ------
+    ValueError: If the backend does not implement ``numpy.scipy.fft``
+
+    """
+    backend = _backend_from_arg(backend)
+    ua.register_backend(backend)
+
+
+def set_backend(backend, coerce=False, only=False):
+    """Context manager to set the backend within a fixed scope.
+
+    Upon entering the ``with`` statement, the given backend will be added to
+    the list of available backends with the highest priority. Upon exit, the
+    backend is reset to the state before entering the scope.
+
+    Parameters
+    ----------
+    backend: {object, 'scipy'}
+
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'}, or an object that implements the uarray protocol.
+
+    coerce: bool, optional
+
+        Whether to allow expensive conversions for the ``x`` parameter. e.g.
+        copying a numpy array to the GPU for a CuPy backend. Implies ``only``.
+
+    only: bool, optional
+
+       If only is ``True`` and this backend returns ``NotImplemented`` then a
+       BackendNotImplemented error will be raised immediately. Ignoring any
+       lower priority backends.
+
```

Comment:
There probably should be a `Returns` section here, to make clear it returns a context manager

Reply:
Can you point me to an example for this? I don't really want to name the type so should I just say
```
Returns
-------
out : context manager
```

Reply:
Context managers typically return something when you call them, for example `with warnings.catch_warnings(record=True) as w:` returns a list of warnings emitted thus far. The question is, what does this actually return? From [here](https://uarray.readthedocs.io/en/latest/_modules/uarray/_backend.html#set_backend) it looks like it returns whatever is returned by `_SetBackendContext`.

I'm not sure what this *does* return, but maybe we should decide something useful that it *should* return, and return that. Like maybe a tuple of backends it will try to use in order of preference? Maybe this should be a separate issue for now, though, since it might require discussion.

Reply:
This is meant to be used as follows:

```python
with set_backend(backend):
    # code here uses your backend
```

There's also `set_global_backend` and `register_backend`.

Reply:
Okay maybe let's leave out the `Returns` section for now, then, until we make `set_backend` return something useful. The `scipy.special.errstate` does not specify any Returns value so this is also good for consistency:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.errstate.html

Sorry @peterbell10 to steer you wrong!

Reply:
Alright I've removed the `Returns` section again. Good to know for sure either way.